### ISSUE TITLE
It should be easy to just clone and run samples

### ISSUE THREAD

**Issue by amanzan** [eneim/toro#316]
First of all, thanks for the work. In my opinion, there should be changes in your repo to be able to clone it and run the samples with no hassle. Right now when you do it, you get multiple errors of needed youtube_apikey, fabric key, and some others, and it's quite complicated to make it run. This discourages people to test your library. Just something you should consider :)

**Comment by eneim** [comment]
@amanzan Thanks, this is one of the things I'm trying to improve, with kind of check for empty fields at build time.

**Comment by eneim** [comment]
@amanzan I have pushed the update to dev-v3. Only thing you should care about is to use Android Studio 3.1.0 (RC3+). Can you give it a try?

**Comment by amanzan** [comment]
@eneim I upgraded to Android Studio 3.1.0 RC3 and I was getting this error after cloning: "Error:Could not find method crashlytics() for arguments [build_8xwqzk0zvu5p0umlalhr1pw55$_run_closure3@b0d35bc] on project ':app-youtube' of type org.gradle.api.Project."

Removing crashlytics lines from build.gradle in app-youtube fixes this issue.

Then I get this warnings: 
```
Folder /Users/server/Desktop/android/toro/app/build/generated/source/kaptKotlin/debug


Folder /Users/server/Desktop/android/toro/app/build/generated/source/kaptKotlin/release


Folder /Users/server/Desktop/android/toro/demo-exoplayer/build/generated/source/kaptKotlin/debug


Folder /Users/server/Desktop/android/toro/demo-exoplayer/build/generated/source/kaptKotlin/release


3rd-party Gradle plug-ins may be the cause
```
I was able to run the app module but it freezes while playing a video (I can hear the audio) so I don't know if the previous warnings have something to do with it

Thanks for the effort :)

**Comment by eneim** [comment]
@amanzan Thanks for the try. Yeah I will take a look on the "crashlytic" block.

About the warnings, I doubt it is something about kotlin-kapt. It is no harm, but doesn't look good. Though you can try ``./gradlew clean`` to see if the warnings are gone.

The freezes may depend on other factor. If you can send me the detail about what device you use, Android version, video Url (if you made any custom on them) I will take a look. Sometime old device doesn't support a certain codec and avc profile that cause the frame to not be rendered.

**Comment by amanzan** [comment]
Yes I think freezing could be something unrelated to the build. 

I run it again and now I am able to see the samples. Thanks, because I'm struggling to get smooth scrolling in a RecyclerView with videos, so I'll give your library a try. 

One thing I noticed (it happens with all other libraries I used and I think with yours too) is that on flinging, at the exact moment the video gets off screen, there's a very small "stop" or "hiccup" and the flinging is not completely smooth. It's not super noticeable but it bothers me a little :) Do you have any idea how to fix it?

Another question, in regards to scrolling performance, is it better to use the ExoPlayer version or just the normal one (I assume this extends somehow from TextureView)?

**Comment by eneim** [comment]
@amanzan It bothers me too, but sadly it is kind of unavoidable ... I will try to see if I can reduce the bad UX, but let's think it is difficult and low priority.

About the last question, by saying "the normal one" you mention the "MediaPlayer"? I have not tried it aggressively, but it has the issue of supporting many Android version. Also using TextureView is proved to improve the smooth scroll on <21, but on 21 and above it is suggested to use SurfaceView again. Also using TextureView requires hardware acceleration which is told to be power consuming.
